design/handle-backup-of-volumes-by-resources-filters.md

```diff
@@ -0,0 +1,246 @@
+
+A new variable `resourcePolices` would be added into `BackupSpec`, it's value is assigned with the current resources policy configmap
+```yaml
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: backup-1
+  spec:
+    resourcePolices:
+      refType: Configmap
+      ref: backup01
+    ...
+```
+
+The configmap basically equivalent to what is generated by the command `kubectl create cm backup01 --from-file policies.yaml`
+ 
+The configmap only stores those assigned values, not the whole resources polices.
+ 
+The name of the configmap is `$BackupName`, and it's in Velero install namespace.
+ 
+#### Life-cycle of resources policies configmap
+- Resource policies configmap only been generated once Velero backup command with flag `backup-policy-configmap` to import configuration.
+- Resource policies configmap could be referenced by another backup in Velero backup command with the flag `policy-from-backup` to reference configuration in another backup.
+- If all referenced backup has been deleted, the relevant configmap should be removed.
+#### Versioning
+Here we introduced the version field in the YAML config to contain break changes in case of some compatibility problems:
+```yaml
+version: v1
+volumePolicies:
+  ....
+```
+#### Multiple versions supporting
+To manage the effort for maintenance, we will only support one version of the data in velero. Suppose that there is one break change for the YAML in Velero v1.13, we should bump up the config version to V2, and v2 is only supported in v1.13. For the existing data with version: v1, it should migrate them when the Velero startup, this won't hurt the existing backup schedule CR as it only references the configmap. To make the migration easier, the configmap for such resource filter policies should be labeled manually before Velero startup like this, Velero will migrate the labeled configmap.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+# This label can be optional but if this is not set, the backup will fail after the breaking change and the user will need to update the data manually 
+    velero.io/resource-filter-policies: "true"
+  name: example
+  namespace: velero
+data:
+  .....
+```
+### Display of resources polices
+As the resource policies configmap is referenced by backup CR, the policies in configmap are not so intuitive, so we need to integrate policies in configmap to the output of the command `velero backup describe`, and make it more readable.
+ 
+## Compatibility
+Currently, we have these resources filters:
+- IncludedNamespaces
+- ExcludedNamespaces
+- IncludedResources
+- ExcludedResources
+- LabelSelector
+- OrLabelSelectors
+- IncludeClusterResources
+- UseVolumeSnapshots
+- velero.io/exclude-from-backup=true
+- backup.velero.io/backup-volumes-excludes
+- backup.velero.io/backup-volumes
+- backup.velero.io/must-include-additional-items
+ 
+So it should be careful with the combination of volumes resources policies and the above resources filters.
```

Comment:
What is the expected behavior if a volume matches policy with action "volume-snapshot" but the backup spec has defaultVolumesToRestic?

Reply:
every policy has related action, if the action is a `volume-snapshot` but we are doing backup with Resitc, then this policy will not take effect.

Reply:
IMO the actions like "volume-snapshot" are put here to show the extensibility of this design, and in v1.11 the MVP is to support `skip`

We'll need to specify the priority when there's conflict when introducing more types of `actions`, thoughts?

Reply:
Makes sense. From Qiuming's comment I guess the policy is given less priority in all cases.